extensions/markdown-language-features/package.json

```diff
@@ -191,6 +196,10 @@
         {
           "command": "_markdown.copyImage",
           "when": "webviewId == 'markdown.preview' && webviewSection == 'image'"
+        },
+        {
+          "command": "_markdown.openImage",
```

Comment:
This command should likely only be enabled for images that can be opened locally. For example, images that are part of the workspace

For this, you'll need to add a new `webviewSection` called something like `localImage` that is only set if the image is local. Take a look at how the contexts are set here:

https://github.com/microsoft/vscode/blob/5ee455116f5399293a854129299ed917ee8d4665/extensions/markdown-language-features/preview-src/index.ts#L129